staging/src/k8s.io/apiserver/pkg/server/healthz/healthz.go

```diff
@@ -255,8 +257,14 @@ func handleRootHealth(name string, firstTimeHealthy func(), checks ...HealthChec
 		}
 		// always be verbose on failure
 		if len(failedChecks) > 0 {
+			// update the stacktrace predicate so that httplog's output is condensed
+			// this handler is used for health checking on the paths like /healthz or /readyz
+			// for some (/readyz) the StatusInternalServerError is considered as a "normal" response code to signal the other end
+			if disableStacktraceForHttpLog {
+				httplog.DisableStackTraceForRequest(r)
+			}
 			klog.V(2).Infof("%s check failed: %s\n%v", strings.Join(failedChecks, ","), name, failedVerboseLogOutput.String())
-			http.Error(httplog.Unlogged(r, w), fmt.Sprintf("%s%s check failed", individualCheckOutput.String(), name), http.StatusInternalServerError)
+			http.Error(w, fmt.Sprintf("%s%s check failed", individualCheckOutput.String(), name), http.StatusInternalServerError)
```

Comment:
why this change?

Reply:
to capture the status code and put it in the log file.

`httplog.Unlogged` returns the original ResponseWriter, or w if it is not our inserted logger.